src/main/java/com/exasol/util/AbstractBottomUpTreeNode.java

```diff
@@ -99,6 +99,16 @@ public boolean isChild() {
         return (this.parent != null);
     }
 
+    @Override
+    public boolean isSibling(final TreeNode node) {
+        return (this.parent != null) && (this.getParent().getChildren().contains(node));
+    }
+
+    @Override
```

Comment:
Should be in `AbstractTreeNode`.

Reply:
`AbstractTreeNode` and `AbstractBottomUpTreeNode` are two independent classes, so I have to implement methods twice. I'm also a little bit puzzled which one I should use in which case...